backend/drm/drm.c

```diff
@@ -700,6 +700,13 @@ static bool drm_connector_set_cursor(struct wlr_output *output,
 
 		swap_drm_surface_buffers(&plane->surf, NULL);
 
+		// workaround for nouveau
+		// Buffers created with GBM_BO_USER_LINEAR are placed in NOUVEAU_GEM_DOMAIN_GART.
+		// When the bo is attached to the cursor plane it is moved to NOUVEAU_GEM_DOMAIN_VRAM.
+		// However, this does not wait for the render operations to complete, leaving an empty surface.
+		// The render operations can be waited for using:
+		glFinish();
```

Comment:
Is this a nouveau bug?

Reply:
Yes, but very unlikely to be fixed if we don't do it

Reply:
Why's that? Have you reported it upstream?

Reply:
The usecase is quite niche (I checked other compositors, they all roundtrip trough the cpu). I talked to the nouveau devs on freenode and that's what they said. I have yet to make a bug report so that it is documented.

Reply:
Have you considered digging into nouveau and writing a patch for them?

Reply:
I definitely have, I already had to dig into nouveau to find the reason for the failure. However, I can't promise that I can do this is in the near future.  I see this more as a stop-gap.

Reply:
Okay. Well, file the bug with freedesktop at least and link it in the comments, and I'll merge this patch.